numba/tests/test_function.py

```diff
@@ -0,0 +1,575 @@
+                         njit2_func]:
+                # Skip not-yet-supported functions
+                continue
+            with self.subTest(decor=decor.__name__):
+                a_ = decor(a)
+                b_ = decor(b)
+                self.assertEqual(njit(foo)(a_, b_, True).pyfunc,
+                                 foo(a, b, True))
+                self.assertEqual(njit(foo)(a_, b_, False).pyfunc,
+                                 foo(a, b, False))
+                self.assertNotEqual(njit(foo)(a_, b_, True).pyfunc,
+                                    foo(a, b, False))
+
+    def test_in_choose_func_value(self):
+
+        def a(i):
+            return i + 1
+
+        def b(i):
+            return i + 2
+
+        def foo(a, b, choose_left):
+            if choose_left:
+                f = a
+            else:
+                f = b
+            return f(1)
+
+        for decor in all_func_kinds:
+            if decor in [pure_func, ctypes_func, njit_func, njit2_func]:
+                # Skip not-yet-supported functions
+                continue
+            with self.subTest(decor=decor.__name__):
+                a_ = decor(a)
+                b_ = decor(b)
+                self.assertEqual(njit(foo)(a_, b_, True), foo(a, b, True))
+                self.assertEqual(njit(foo)(a_, b_, False), foo(a, b, False))
+                self.assertNotEqual(njit(foo)(a_, b_, True), foo(a, b, False))
+
+    def test_in_pick_func_call(self):
+
+        def a(i):
+            return i + 1
+
+        def b(i):
+            return i + 2
+
+        def foo(funcs, i):
+            r = funcs[i](123)
+            return r
+
+        for decor in all_func_kinds:
+            if decor in [pure_func, ctypes_func, njit_func, njit2_func]:
+                # Skip not-yet-supported functions
+                continue
+            with self.subTest(decor=decor.__name__):
+                a_ = decor(a)
+                b_ = decor(b)
+                self.assertEqual(njit(foo)((a_, b_), 0), foo((a, b), 0))
+                self.assertEqual(njit(foo)((a_, b_), 1), foo((a, b), 1))
+                self.assertNotEqual(njit(foo)((a_, b_), 0), foo((a, b), 1))
+
+    def test_in_iter_func_call(self):
+
+        def a(i):
+            return i + 1
+
+        def b(i):
+            return i + 2
+
+        def foo(funcs, n):
+            r = 0
+            for i in range(n):
+                f = funcs[i]
+                r = r + f(r)
+            return r
+
+        for decor in all_func_kinds:
+            if decor in [pure_func, ctypes_func, njit_func, njit2_func]:
+                # Skip not-yet-supported functions
+                continue
+            with self.subTest(decor=decor.__name__):
+                a_ = decor(a)
+                b_ = decor(b)
+                self.assertEqual(njit(foo)((a_, b_), 2), foo((a, b), 2))
+
+
+class TestFuncionTypeExtensions(TestCase):
+
+    def _test_wrapper_address_protocol(self):
```

Comment:
I feel that a variant of this test which talked to a `libm` function would be more convincing in that it can be asserted a known value is returned (for example, calling `libm::cos(0)`) opposed to relying on derived properties.

Reply:
I have added a new WAP test that calls `cos` and `sinf` from `libm`.

Enabling `time` test. However, IIRC, it failed in one of the CI nodes.